### PR TITLE
[WIP] This adds the buildkite setup

### PR DESCRIPTION
**Pull request by ven-k**
- Checks if the folder in `benchmarks` has `.jmd` file
- Creates a pipeline for each `.jmd` file
- If there is any change in the rest of the repo runs the tests
- Uploads pdf, html... as artifacts

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Changes:

- Make trigger like https://github.com/SciML/SciMLBenchmarks.jl/issues/116
- Setup with the AMDCI3 tags
- Copy same thing over to SciMLTutorials
- Allow marking the GPU benchmarks to use gpuci1

**Comment by ChrisRackauckas** [comment]
@staticfloat mentioned we should go away from text triggers. What do you recommend? Our biggest issue here is that these benchmarks should hardly ever be updated, but should be run often.

**Comment by ven-k** [comment]
We can have a workflow dispatch ([a demo at 1 min](https://dev.to/github/github-actions-manual-triggers-with-workflowdispatch-3di1) - this can be modified to accept (folder, file) options)

That's one of the 2 ways to manually trigger the workflow through Github actions. (The other one is `repository_dispatch` for integration if there are GUIs). This workflow can have buildkite pipeline (maybe[ this](https://github.com/marketplace/actions/trigger-buildkite-pipeline#:~:text=Usage,the%20settings%20for%20the%20build) plugin is useful there) 

Buildkite's options to trigger didn't seem this flexible (considering "no-text" triggering)

File: .github/workflows/update.jl [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    cd(model_dir)
+    Pkg.activate(".")
+    Pkg.update()
+    Pkg.activate()
+    manpath = (joinpath(benchpath, "Manifest.toml"))
+    if length(readlines(`git status . --porcelain`)) > 1
+        run(`git checkout -B $dir`)
+        run(`$git add -A . :!manpath`)
+        run(`$git commit -m "Updated $dir"`)
+        run(`$git push --set-upstream origin $dir`)
```
Comment (ven-k) [inline .github/workflows/update.jl:21]: Would `git request-pull $dir origin master` create a PR?

**Comment by ven-k** [comment]
@staticfloat as "Request a reviewer" isn't showing up, I'm tagging you for a review.

File: .buildkite/rebuild_models.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+steps:
+  - label: ":runner: Dynamically launch Pipelines"
+    plugins:
+      - staticfloat/forerunner:
+          # This will create one job per file
+          watch:
+            - path: benchmarks/.*\.jmd
+          path_processor: per-file
+          target: .buildkite/coalescer/rebuild_models
+          target_type: template
```
Comment (staticfloat) [inline .buildkite/rebuild_models.yml:13]: Can you explain to me what you're trying to do here?

Comment (ven-k) [reply .buildkite/rebuild_models.yml:13]: I want to watch the `.jmd` files. If there is any modification then run the benchmarks for just that file.

The `.buildkite/coalescer/rebuild_models` currently creates a `MODELS` which has "folder_A/modified-file.jmd" as entries. Currently, the script only does an `echo` (should I do something to pass this) and it is supposed to be used by `rebuild_models.jl` to `weave_file(folder, file)`.

Earlier, something like `!rebuild folder_A` would make it check for the `modified.jmd` and launch a pipeline for each of them.

File: .buildkite/commands/rebuild_models.jl [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+model = ARGS[1]
+folder, file = split(model, "/")
+
+project_abspath = joinpath(repo_dir, "library", model)
+if !isfile(project_abspath) || !occursin(file, ".jmd")
+    error("Invalid file $file")
+end
+
+@info("Rebuilding $folder/$file")
+weave_folder(folder, file)
```
Comment (ven-k) [inline .buildkite/commands/rebuild_models.jl:15]: This has to be `weave_file(folder, file)`. Will fix it in a subsequent commit

**Comment by ChrisRackauckas** [comment]
Resolve conflicts and let's merge. Let's just see what happens.

**Pull request merged**